src/fidesops/service/authentication/authentication_strategy_basic.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Dict
+
+from requests import PreparedRequest
+
+from fidesops.schemas.saas.strategy_configuration import (
+    BasicAuthenticationConfiguration,
+    StrategyConfiguration,
+)
+from fidesops.service.authentication.authentication_strategy import (
+    AuthenticationStrategy,
+)
+from fidesops.util.saas_util import assign_placeholders
+
+STRATEGY_NAME = "basic"
+
+
+class BasicAuthenticationStrategy(AuthenticationStrategy):
+    """
+    Replaces the username and password placeholders with the actual credentials
+    and uses them to add a basic authentication header to the incoming request.
+    """
+
+    def __init__(self, configuration: BasicAuthenticationConfiguration):
+        self.username = configuration.username
+        self.password = configuration.password
+
+    def get_strategy_name(self) -> str:
+        return STRATEGY_NAME
+
+    def add_authentication(
+        self, request: PreparedRequest, secrets: Dict[str, Any]
+    ) -> PreparedRequest:
+        """Add basic authentication to the request"""
+        request.prepare_auth(
+            auth=(
+                assign_placeholders(self.username, secrets),
+                assign_placeholders(self.password, secrets),
+            )
+        )
+        return request
+
+    @staticmethod
+    def get_configuration_model() -> StrategyConfiguration:
+        return BasicAuthenticationConfiguration
```

Comment:
Thinking more about this, we can abstract it away to the `AuthenticationStrategy` class, and have it become automatically available on anything that inherits from it 😎

Reply:
So clean! 😎

Reply:
I tried some things locally to abstract away `get_configuration_model` from each subclass of `AuthenticationStrategy`, but I can see that it won't work because of [this hacky code](https://github.com/ethyca/fidesops/blob/5a5a87a030201163f53925edb89e2e0165400b73/src/fidesops/service/authentication/authentication_strategy_factory.py#L58-L63). The other strategy factories do things this way, so it's fine to leave it for now. A larger refactor for another PR 😕